gradle-plugin/kubernetes/src/main/java/org/eclipse/jkube/gradle/plugin/task/AbstractJKubeTask.java

```diff
@@ -59,22 +90,78 @@ protected String getLogPrefix() {
+                            .passwordDecryptionMethod(s -> s)
+                            .registry(kubernetesExtension.getPullRegistry().getOrElse(kubernetesExtension.getRegistry().getOrElse("docker.io")))
+                            .build())
+                    .build())
+                    .clusterAccess(clusterAccess)
+                    .offline(kubernetesExtension.getOffline().getOrElse(DEFAULT_OFFLINE))
+                    .platformMode(kubernetesExtension.getRuntimeMode());
+    if (isDockerAccessRequired()) {
+      DockerAccessFactory.DockerAccessContext dockerAccessContext = DockerAccessFactory.DockerAccessContext.builder()
+              .log(kitLogger)
+              .projectProperties(javaProject.getProperties())
+              .maxConnections(kubernetesExtension.getMaxConnections().getOrElse(DEFAULT_MAX_CONNECTIONS))
+              .dockerHost(kubernetesExtension.getDockerHost().getOrNull())
+              .certPath(kubernetesExtension.getCertPath().getOrNull())
+              .machine(kubernetesExtension.machine)
+              .minimalApiVersion(kubernetesExtension.getMinimalApiVersion().getOrNull())
+              .skipMachine(kubernetesExtension.getSkipMachine().getOrElse(false))
+              .build();
+      DockerAccessFactory dockerAccessFactory = new DockerAccessFactory();
+      ImagePullManager imagePullManager = createImagePullManager(
+              kubernetesExtension.getImagePullPolicy().getOrElse("Always"),
+              kubernetesExtension.getAutoPull().getOrElse("true"),
+              javaProject.getProperties());
+      DockerAccess access = dockerAccessFactory.createDockerAccess(dockerAccessContext);
+      ServiceHubFactory serviceHubFactory = new ServiceHubFactory();
+      LogOutputSpecFactory logSpecFactory = new LogOutputSpecFactory(true, true, null);
+      builder.dockerServiceHub(serviceHubFactory.createServiceHub(access, kitLogger, logSpecFactory));
+      builder.buildServiceConfig(BuildServiceConfig.builder()
+                      .buildRecreateMode(BuildRecreateMode.fromParameter(kubernetesExtension.getBuildRecreate().getOrElse("none")))
+                      .jKubeBuildStrategy(kubernetesExtension.getBuildStrategy().getOrElse(JKubeBuildStrategy.docker))
+                      .forcePull(kubernetesExtension.getForcePull().getOrElse(false))
+                      .buildDirectory(javaProject.getBuildDirectory().getAbsolutePath())
+                      .imagePullManager(imagePullManager)
+                      .build());
+    }
+
+    return builder;
   }
 
   protected ClusterConfiguration initClusterConfiguration() {
     return ClusterConfiguration.from(kubernetesExtension.access,
         System.getProperties(), javaProject.getProperties()).build();
   }
 
+  // Exposed for testing
+  protected JKubeServiceHub initJKubeServiceHub() {
+    return initJKubeServiceHubBuilder().build();
+  }
```

Comment:
This is redundant of `initJKubeServiceHubBuilder`. Why is it needed?

Reply:
I wanted to use a mocked JKubeServiceHub instance in tests, I could only find this way to override this method and use mocked instance in tests

Reply:
OK, we need to see if this can be addressed in another way